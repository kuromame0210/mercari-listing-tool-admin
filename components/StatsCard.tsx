import { LucideIcon } from 'lucide-react';

interface StatsCardProps {
  title: string;
  value: string;
  icon: LucideIcon;
  color: string;
}

export default function StatsCard({ title, value, icon: Icon, color }: StatsCardProps) {
  return (
    <div className="card">
      <div className="flex items-center">
        <div className="flex-1">
          <p className="text-sm font-medium text-gray-600">{title}</p>
          <p className="text-2xl font-bold text-gray-900">{value}</p>
        </div>
        <div className={`w-12 h-12 ${color} rounded-lg flex items-center justify-center`}>
          <Icon className="h-6 w-6 text-white" />
        </div>
      </div>
    </div>
  );
}