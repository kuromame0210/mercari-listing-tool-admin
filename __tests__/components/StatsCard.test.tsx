import { render, screen } from '@testing-library/react'
import { Package } from 'lucide-react'
import StatsCard from '../../components/StatsCard'

describe('StatsCard Component', () => {
  const defaultProps = {
    title: 'テスト統計',
    value: '1,234',
    icon: Package,
    color: 'bg-blue-500'
  }

  test('正しくレンダリングされる', () => {
    render(<StatsCard {...defaultProps} />)
    
    expect(screen.getByText('テスト統計')).toBeInTheDocument()
    expect(screen.getByText('1,234')).toBeInTheDocument()
  })

  test('タイトルが表示される', () => {
    render(<StatsCard {...defaultProps} title="商品数" />)
    
    expect(screen.getByText('商品数')).toBeInTheDocument()
  })

  test('値が表示される', () => {
    render(<StatsCard {...defaultProps} value="999" />)
    
    expect(screen.getByText('999')).toBeInTheDocument()
  })

  test('アイコンが表示される', () => {
    render(<StatsCard {...defaultProps} />)
    
    // Lucide ReactのPackageアイコンがSVGとしてレンダリングされることを確認
    const icon = document.querySelector('svg')
    expect(icon).toBeInTheDocument()
  })

  test('背景色クラスが適用される', () => {
    render(<StatsCard {...defaultProps} color="bg-green-500" />)
    
    const iconContainer = document.querySelector('.bg-green-500')
    expect(iconContainer).toBeInTheDocument()
  })

  test('大きな数値が正しく表示される', () => {
    render(<StatsCard {...defaultProps} value="1,000,000" />)
    
    expect(screen.getByText('1,000,000')).toBeInTheDocument()
  })

  test('長いタイトルが表示される', () => {
    const longTitle = 'とても長い統計項目のタイトルです'
    render(<StatsCard {...defaultProps} title={longTitle} />)
    
    expect(screen.getByText(longTitle)).toBeInTheDocument()
  })

  test('空の値でも正しく動作する', () => {
    render(<StatsCard {...defaultProps} value="" />)
    
    expect(screen.getByText('テスト統計')).toBeInTheDocument()
  })

  test('異なる色の組み合わせで正しく動作する', () => {
    const colors = ['bg-red-500', 'bg-yellow-500', 'bg-purple-500']
    
    colors.forEach(color => {
      const { container } = render(<StatsCard {...defaultProps} color={color} />)
      const iconContainer = container.querySelector(`.${color}`)
      expect(iconContainer).toBeInTheDocument()
    })
  })

  test('スナップショット', () => {
    const { container } = render(<StatsCard {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})