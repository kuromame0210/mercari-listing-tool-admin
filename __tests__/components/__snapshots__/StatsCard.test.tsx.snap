// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard Component スナップショット 1`] = `
<div
  class="card"
>
  <div
    class="flex items-center"
  >
    <div
      class="flex-1"
    >
      <p
        class="text-sm font-medium text-gray-600"
      >
        テスト統計
      </p>
      <p
        class="text-2xl font-bold text-gray-900"
      >
        1,234
      </p>
    </div>
    <div
      class="w-12 h-12 bg-blue-500 rounded-lg flex items-center justify-center"
    >
      <svg
        class="lucide lucide-package h-6 w-6 text-white"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m7.5 4.27 9 5.15"
        />
        <path
          d="M21 8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16Z"
        />
        <path
          d="m3.3 7 8.7 5 8.7-5"
        />
        <path
          d="M12 22V12"
        />
      </svg>
    </div>
  </div>
</div>
`;
